### DISCUSSION TITLE
Added Support for CloudFlare Tunnels

### DISCUSSION THREAD

**Discussion by TheMiningAlex** [pelican-dev/panel#22] [Ideas]
Currently on my version of Pterodactyl, It is not possible to run the panel or wings through a CloudFlare Tunnel, If possible this would be a great featutre addition to Pelican!

**Comment by purejosh** [comment]
Are you referring to Cloudflare's "proxied" DNS feature?

**Reply by wdhdev** [reply]
I think he's referring to [cloudflared](https://github.com/cloudflare/cloudflared).

**Reply by dxomg** [reply]
But the thing is, it works just fine for me on cloudflared, both panel and wings

**Reply by KeyTerminal** [reply]
I am currently using Cloudflare's Tunnel myself without issue.

**Reply by NotShocked** [reply]
Currently on pterodactyl if you set wings to use SSL AND behind a proxy it turns SSL off when wings are restarted.

This is 100% reproducible on multiple wings nodes. Yet its a valid setup to have. 

I think many people do that setup and then think it doesn't work because they are attempting a HTTPS connection and wings is only accepting HTTP.

**Reply by Korenchkin** [reply]
maybe it is the same,but add 'me too' for support of 'behind ssl' a.k.a 1 ipv4 address,nat router->A) caddy/other proxy with letsencrypt->panel+wings web/api and  B) direct nat to other wing(s) game ports
now in pterodactyl it works,but you have to cheat wings config,it is undocumented,it seems everybody expects you to have multiple public ipv4 addresses,but for home selfhosters one is enough and you can't(/don't want to) really work witthout https  :)
also could be single machine/same ip

**Comment by lancepioch** [comment]
Yes we are considering it!

**Comment by lancepioch** [comment]
From the testing I've seen the Panel is working with cloudflared!
